Public/Src/Tools/Execution.Analyzer/Analyzers.Partner/ContentPlacementAnalyzer.cs

```diff
@@ -0,0 +1,532 @@
+            else
+            {
+                dictionary.Add(key, new List<ValType>(new ValType[] { value }));
+            }
+        }
+
+        private void CountInDictionary<KeyType>(Dictionary<KeyType, int> dictionary, KeyType key)
+        {
+            if (dictionary.ContainsKey(key))
+            {
+                dictionary[key] += 1;
+            }
+            else
+            {
+                dictionary.Add(key, 1);
+            }
+        }
+
+    }
+
+    public sealed class BXLPipData
+    {
+        public string SemiStableHash { get; set; } = null;
+        public int Priority { get; set; } = 0;
+        public int Weight { get; set; } = 0;
+        public int TagCount { get; set; } = 0;
+        public int SemaphoreCount { get; set; } = 0;
+        public double DurationMs { get; set; } = -1;
+        public string Type { get; set; } = null;
+        public string ExecutionLevel { get; set; } = null;
+        public int DependencyCount { get; set; } = 0;
+        public int InputCount { get; set; } = 0;
+        public int OutputCount { get; set; } = 0;
+        public long StartTimeTicks { get; set; } = 0;
+
+        public static List<BXLPipData> BuildFromExecutionData(Pip pip, IList<PipExecutionPerformanceEventData> executionData,
+            Dictionary<uint, int> scheduledPipDependencyCount,
+            Dictionary<uint, int> scheduledPipInputCount,
+            Dictionary<uint, int> scheduledPipOutputCount,
+            PathTable pathTable,
+            StringTable stringTable)
+        {
+            var output = new List<BXLPipData>();
+            foreach (var eData in executionData)
+            {
+                output.Add(BuildFromExecutionData(pip, eData, scheduledPipDependencyCount, scheduledPipInputCount, scheduledPipOutputCount, pathTable, stringTable));
+            }
+            return output;
+        }
+
+        public static BXLPipData BuildFromExecutionData(Pip pip, PipExecutionPerformanceEventData executionData,
+            Dictionary<uint, int> scheduledPipDependencyCount,
+            Dictionary<uint, int> scheduledPipInputCount,
+            Dictionary<uint, int> scheduledPipOutputCount,
+            PathTable pathTable,
+            StringTable stringTable)
+        {
+            var pipData = new BXLPipData();
+            pipData.SemiStableHash = pip.FormattedSemiStableHash;
+            pipData.DurationMs = (executionData.ExecutionPerformance.ExecutionStop - executionData.ExecutionPerformance.ExecutionStart).TotalMilliseconds;
+            pipData.Type = pip.PipType.ToString();
+            pipData.ExecutionLevel = executionData.ExecutionPerformance.ExecutionLevel.ToString();
+            pipData.StartTimeTicks = executionData.ExecutionPerformance.ExecutionStart.Ticks;
+            pipData.TagCount = pip.Tags.Count();
+            int depCount;
+            int iCount;
+            int oCount;
+            scheduledPipDependencyCount.TryGetValue(pip.PipId.Value, out depCount);

```

Comment:
>out depCoun [](start = 69, length = 11)

nit: We have enabled the latest C# features. You can now write a more condensed version:
`out var depCount`
so you can remove the `int depCount` line.